changes.d/6354.feat.md

```diff
@@ -0,0 +1 @@
+Fix an issue that could cause clock-expired tasks to be erroneously retried if "execution retry delays" were configured.
```

Comment:
This doesn't match the issue description in the PR?

Reply:
Oh heck!

Reply:
Fixed.